packages/store-sync/src/indexer-client/createIndexerClient.ts

```diff
@@ -0,0 +1,42 @@
+import { z } from "zod";
+import { input } from "./input";
+import { StorageAdapterBlock } from "../common";
+import { Result } from "@latticexyz/common";
+
+type CreateIndexerClientOptions = {
+  /**
+   * tRPC endpoint URL like `https://indexer.dev.linfra.xyz/trpc`.
+   */
+  url: string;
+};
+
+type IndexerClient = {
+  getLogs: (opts: z.input<typeof input>) => Promise<Result<StorageAdapterBlock>>;
+};
+
+/**
+ * Creates a client to talk to a MUD indexer.
+ *
+ * @param {CreateIndexerClientOptions} options See `CreateIndexerClientOptions`.
+ * @returns {IndexerClient} A typed indexer client.
+ */
+export function createIndexerClient({ url }: CreateIndexerClientOptions): IndexerClient {
+  return {
+    getLogs: async (opts): Promise<Result<StorageAdapterBlock>> => {
+      const input = encodeURIComponent(JSON.stringify(opts));
+      const response = await fetch(`${url}/get/logs?input=${input}`, { method: "GET" });
+
+      // TODO: could we return a readable stream here instead of fetching the entire response right away?
+      const result = await response.json();
+      if (!isStorageAdapterBlock(result)) {
+        return { error: "Unexpected response:\n" + JSON.stringify(result) };
+      }
+
+      return { ok: { ...result, blockNumber: BigInt(result.blockNumber) } };
```

Comment:
should we wrap this whole thing in a try/catch to return error type? like if the fetch fails or stringifying opts (i.e. bigint)